packages/contracts-bedrock/src/L1/OptimismPortalInterop.sol

```diff
@@ -58,4 +116,55 @@ contract OptimismPortalInterop is OptimismPortal2 {
             )
         );
     }
+
+    /// @notice Getter for the address of the shared lockbox.
+    function sharedLockbox() external view returns (ISharedLockbox) {
+        return ISharedLockbox(_storage().sharedLockbox);
+    }
+
+    /// @notice Getter for the migrated flag.
+    function migrated() external view returns (bool) {
+        return _storage().migrated;
+    }
+
+    /// @notice Unlock and receive the ETH from the SharedLockbox.
+    /// @param _tx Withdrawal transaction to finalize.
+    function _unlockETH(Types.WithdrawalTransaction memory _tx) internal virtual override {
+        OptimismPortalStorage storage s = _storage();
+
+        // We don't allow the SharedLockbox to be the target of a withdrawal.
+        // This is to prevent the SharedLockbox from being drained.
+        // This check needs to be done for every withdrawal.
+        if (_tx.target == s.sharedLockbox) revert MessageTargetSharedLockbox();
+
+        if (!s.migrated) return;
```

Comment:
It would be good to add some comments about these return statements as well. If it hasn't been migrated, we keep the same behavior as the existing portal where the ether accumulates in the portal rather than the lockbox. If the value is 0 then its just an optimization in saving a call